hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/distribution/DistributionTraitDef.java

```diff
@@ -97,6 +97,10 @@ public RelNode convert(
             return rel;
         }
 
+        if (currentTrait.satisfies(targetTrait)) {
```

Comment:
With the new `satisfies` check, the upper `equals` check is no longer needed.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/18013/commits/7d123a63b095335ec190e28d54c4bce3488a4a02